Segment fault In renderCursor of Textfield

### Describe the bug

```
func (tf *TextField) renderCursor(on bool) {
	if tf == nil || tf.This == nil {
		return
	}
	if !on {
		if tf.Scene == nil {
			return
		}
// ------ add this checking
		if tf.Scene.Stage == nil {
			return
		}
// ------
		ms := tf.Scene.Stage.Main
		if ms == nil {
			return
		}
		spnm := fmt.Sprintf("%v-%v", textFieldSpriteName, tf.lineHeight)
		ms.Sprites.InactivateSprite(spnm)
		return
	}
	if !tf.IsVisible() {
		return
	}

	tf.cursorMu.Lock()
	defer tf.cursorMu.Unlock()

	sp := tf.cursorSprite(on)
	if sp == nil {
		return
	}
	sp.Geom.Pos = tf.charRenderPos(tf.cursorPos, true).ToPointFloor()
}
```

### How to reproduce

Steps:
- go run main.go

and panic.

### Example code

```go
package main

import (
	"strings"

	"cogentcore.org/core/core"
	"cogentcore.org/core/events"
	"cogentcore.org/core/icons"
)

func main() {
	// 初始化 Cogent Core 应用
	b := core.NewBody("SSH Key Mgr").SetTitle("Cogent Core List Example")
	// 主窗口

	// 1. 顶部工具栏（Toolbar）
	tb := core.NewToolbar(b)
	addBtn := core.NewButton(tb).SetIcon(icons.Add).SetTooltip("Add new item")
	addBtn.OnClick(func(e events.Event) {
		core.MessageSnackbar(b, "Add button clicked!") // 示例：点击后显示消息
	})

	// 2. 中间搜索栏（SearchBar）
	searchBar := core.NewTextField(b).SetPlaceholder("Search...")

	// 3. 动态列表（List）
	list := core.NewList(b) // 创建一个列表

	// 示例数据
	items := []string{"Apple", "Banana", "Cherry", "Date", "Elderberry"}

	// 更新列表内容
	updateList := func() {
		list.DeleteChildren() // 清空现有列表项

		var newitems []string
		searchText := strings.ToLower(searchBar.Text())
		for _, item := range items {
			if searchText == "" || strings.Contains(strings.ToLower(item), searchText) {
				// 添加列表项
				newitems = append(newitems, item)
				// li.OnClick(func(e events.Event) {
				// 	// 点击列表项时弹出窗口
				// 	d := core.NewBody("Item Details")
				// 	tf := core.NewTextField(d).SetText(item)
				// })
			}
		}
		list.SetSlice(newitems)
	}

	// 初始加载列表
	updateList()

	// 搜索栏输入变化时重新过滤列表
	searchBar.OnChange(func(e events.Event) {
		updateList()
	})

	// 运行窗口
	b.RunMainWindow()
}
```

### Relevant output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x78 pc=0x1050d9544]

goroutine 1 [running, locked to thread]:
cogentcore.org/core/core.(*TextField).renderCursor(0x1400027dc08, 0x80?)
    /Users/mac/go/pkg/mod/cogentcore.org/core@v0.3.12-0.20250519050645-90393ed5cf92/core/textfield.go:1247 +0x44
cogentcore.org/core/core.(*TextField).clearCursor(0x1400027dc08)
    /Users/mac/go/pkg/mod/cogentcore.org/core@v0.3.12-0.20250519050645-90393ed5cf92/core/textfield.go:1227 +0x6c
cogentcore.org/core/core.(*TextField).editDone(0x1400027dc08)
    /Users/mac/go/pkg/mod/cogentcore.org/core@v0.3.12-0.20250519050645-90393ed5cf92/core/textfield.go:535 +0x98
cogentcore.org/core/core.(*TextField).Text(...)
    /Users/mac/go/pkg/mod/cogentcore.org/core@v0.3.12-0.20250519050645-90393ed5cf92/core/textfield.go:446
main.main.func3(...)
    /Users/mac/repos/golang/sshkeymgr-cogentcore/main.go:40
main.main()
    /Users/mac/repos/golang/sshkeymgr-cogentcore/main.go:56 +0x5a0
exit status 2
```

### Platform

macOS

--- a/core/textfield.go
+++ b/core/textfield.go
@@ -1241,7 +1241,7 @@ func (tf *TextField) renderCursor(on bool) {
 		return
 	}
 	if !on {
-		if tf.Scene == nil {
+		if tf.Scene == nil || tf.Scene.Stage == nil {
 			return
 		}
 		ms := tf.Scene.Stage.Main

